srcs/app/config/analyser/Lexer.cpp

```diff
@@ -2,63 +2,72 @@
 
 Lexer::~Lexer(void) {}
 
-void Lexer::addStringLit(std::string *filebuff, size_t *tokenend) {
+void Lexer::AddStringLit_(std::string *filebuff, size_t *tokenend) {
 	std::string token;
 	char cmp;
 
 	cmp = (*filebuff)[0];
 	*filebuff = filebuff->substr(1);
 	*tokenend = filebuff->find(cmp, 0);
-	if (*tokenend == filebuff->npos)
+	if (*tokenend == std::string::npos)
 		throw Analyser::SyntaxError("Unterminated quote", line_);
 	token = filebuff->substr(0, *tokenend);
 	tokens_.push_back(Token(token, Token::Type::T_WORD, line_));
 	line_ += std::count(token.begin(), token.end(), '\n');
 	(*tokenend)++;
 }
 
-void Lexer::addPunct(char type, size_t *tokenend) {
+void Lexer::AddPunct_(char type, size_t *tokenend) {
 	char			tmp[2];
 	t_token_type	ttype;
 
-	if (type == ';')
-		ttype = Token::Type::T_SEMICOLON;
-	else if (type == '{')
-		ttype = Token::Type::T_SCOPE_OPEN;
-	else if (type == '}')
-		ttype = Token::Type::T_SCOPE_CLOSE;
-	else
-		throw Analyser::SyntaxError("Unexpected token", line_);
+	switch (type) {
+		case ';': {
+			ttype = Token::Type::T_SEMICOLON;
+			break;
+		}
+		case '{': {
+			ttype = Token::Type::T_SCOPE_OPEN;
+			break;
+		}
+		case '}' : {
+			ttype = Token::Type::T_SCOPE_CLOSE;
+			break;
+		}
+		default : {
+			throw Analyser::SyntaxError("Unexpected token", line_);
+		}
+	}
 	tmp[0] = type;
```

Comment:
I've readen in many places that `switch` statements are evil but, as with everything, with moderation and common sense you can use all tools :smile: . I think it fits here :+1: